pkg/tests/end_to_end_tests/trace_query_integration_test.go

```diff
@@ -0,0 +1,396 @@
+	r, err := do(fmt.Sprintf(servicesEndpoint, c.url))
+	require.NoError(t, err)
+
+	arr := convertToStringArr(r.Data.([]interface{}))
+	sort.Strings(arr)
+	return arr
+}
+
+func getOperations(t testing.TB, c httpClient, service string) []string {
+	r, err := do(fmt.Sprintf(operationsEndpoint, c.url, service))
+	require.NoError(t, err)
+
+	arr := convertToStringArr(r.Data.([]interface{}))
+	sort.Strings(arr)
+	return arr
+}
+
+func getTrace(t testing.TB, c httpClient, traceId [16]byte) jaegerJSONModel.Trace {
+	r, err := do(fmt.Sprintf(singleTraceEndpoint, c.url, getTraceId(traceId)))
+	require.NoError(t, err)
+
+	trace := convertToTrace(r.Data.([]interface{})[0])
+	sortTraceContents(&trace)
+	purgeProcessInformation(&trace)
+	return trace
+}
+
+func getTraces(t testing.TB, c httpClient, service string, start, end int64) []jaegerJSONModel.Trace {
+	r, err := do(fmt.Sprintf(fetchTracesEndpoint, c.url, start, end, service))
+	require.NoError(t, err)
+
+	traces := convertToTraces(r.Data.([]interface{}))
+	sort.SliceStable(traces, func(i, j int) bool {
+		return traces[i].TraceID < traces[j].TraceID
+	})
+	for i := range traces {
+		sortTraceContents(&traces[i])
+		purgeProcessInformation(&traces[i])
+	}
+	return traces
+}
+
+func do(url string) (*structuredResponse, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("http get: %w", err)
+	}
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("read all: %w", err)
+	}
+	var r structuredResponse
+	if err = json.NewDecoder(bytes.NewReader(b)).Decode(&r); err != nil {
+		return nil, fmt.Errorf("json unmarshal: %w", err)
+	}
+	return &r, nil
+}
+
+type traceResponse struct {
+	Services   []string
+	Operations []string
+	Trace      jaegerJSONModel.Trace
+	Traces     []jaegerJSONModel.Trace
+}
+
+type traceResponsesStore struct {
+	Responses []traceResponse
+}
+
+const jaegerQueryResponsesPath = "../testdata/jaeger_query_responses.sz"
+
+func loadJaegerQueryResponses() (*traceResponsesStore, error) {
+	f, err := os.Open(jaegerQueryResponsesPath)
+	if err != nil {
+		return nil, fmt.Errorf("opening jaeger query response: %w", err)
+	}
+	var store traceResponsesStore
+	reader := gob.NewDecoder(snappy.NewReader(f))
+	if err = reader.Decode(&store); err != nil {
+		return nil, fmt.Errorf("decoding gob: %w", err)
+	}
+	tweakResults(&store)
+	return &store, nil
+}
+
+// tweakResults modifies trivial stuff that are lost in serialization & deserialization of data.
+func tweakResults(s *traceResponsesStore) {
+	for i := range s.Responses {
+		makeSliceEmptyIfNil(&s.Responses[i].Trace)
+		for j := range s.Responses[i].Traces {
+			makeSliceEmptyIfNil(&s.Responses[i].Traces[j])
+		}
+	}
+}
+
+func makeSliceEmptyIfNil(s *jaegerJSONModel.Trace) {
+	for i := range s.Spans {
+		if s.Spans[i].References == nil {
+			s.Spans[i].References = []jaegerJSONModel.Reference{}
+		}
+		if s.Spans[i].Logs == nil {
+			s.Spans[i].Logs = []jaegerJSONModel.Log{}
+		}
+	}
+}
```

Comment:
Nit: can we put all this into a separate file -- maybe trace_response.go. I would also move 
```
// #nosec
	f, err := os.OpenFile(jaegerQueryResponsesPath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0777)
	require.NoError(t, err)
	defer f.Close()

	require.NoError(t, f.Truncate(0))

	snappyWriter := snappy.NewBufferedWriter(f)

	enc := gob.NewEncoder(snappyWriter)
	require.NoError(t, enc.Encode(&store))
``` 
from `TestGenerateJaegerAPIResponses` into a new function in this new file